pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -152,18 +170,91 @@ func parseURL(input string) (url.URL, error) {
 	return *uri, nil
 }
 
+// getV1Registries transforms the registries arrays of v1 into an array of
+// v2 registries of type Registry.
+func getV1Registries(config *tomlConfig) ([]Registry, error) {
+	regs := make(map[string]Registry)
+
+	makeRegistry := func(s string) (Registry, error) {
+		var reg Registry
+		url, err := parseURL(s)
+		if err != nil {
+			return reg, err
+		}
+		reg.URL = url
+		reg.Prefix = stripURIScheme(url)
+		return reg, nil
+	}
+
+	addRegistry := func(reg Registry) {
+		if up, exists := regs[reg.Prefix]; exists {
+			up.Search = up.Search || reg.Search
+			up.Blocked = up.Blocked || reg.Blocked
+			up.Insecure = up.Insecure || reg.Insecure
+			regs[reg.Prefix] = up
+			return
+		}
+		reg.Mirrors = []Mirror{}
```

Comment:
This line seems to be a better fit with the rest of the intialization in `makeRegistry`

Reply:
Done